packages/components/pagination/src/Pagination.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+
+import { Button } from '@contentful/f36-button';
+import { CommonProps, Stack } from '@contentful/f36-core';
+import { ChevronLeftIcon, ChevronRightIcon } from '@contentful/f36-icons';
+
+export interface PaginationProps extends CommonProps {
+  /**
+   * Sets which page is active on the Pagination
+   */
+  activePage: number;
+  /**
+   * Sets how many items are displayed per page
+   */
+  itemsPerPage: number;
+  /**
+   * Total amount of items the pagination is applied to.
+   */
+  totalItems: number;
+  /**
+   * Handler function called when user navigates to another page on the pagination.
+   */
+  onPageChange: (page: number) => void;
+}
+
+export function getPagesRange(
+  page: number,
+  total: number,
+  neighboursCount = 2,
+): number[] {
+  const PAGINATION_RANGE = neighboursCount * 2;
+
+  if (total <= PAGINATION_RANGE) {
+    // Total amount of pages are less than the possible pagination range
+    return [...Array(total).keys()];
+  }
+
+  const baseRange = [...Array(PAGINATION_RANGE + 1).keys()];
+
+  if (page <= neighboursCount) {
+    // Active page is at the start of the pagination page count
+    return baseRange;
+  }
+  if (page > total - neighboursCount) {
+    // Active page is at the end of the pagination page count
+    return baseRange.map((i) => i + total - PAGINATION_RANGE - 1);
+  }
+  // Active page is in the middle of the pagination count
+  return baseRange.map((i) => i + page - neighboursCount - 1);
+}
```

Comment:
Shouldn't this be in utils as well?

Reply:
Well, I don't know

I think keeping it here and not exporting it is a good idea as well ✌️